gums-core/src/main/java/gov/bnl/gums/account/AccountPoolMapper.java

```diff
@@ -163,6 +169,9 @@ public String getType() {
 
     @Override
     public AccountInfo mapUser(GridUser user, boolean createIfDoesNotExist) {
+        if (groupName != null && !groupName.equals("")) {
+            return new AccountInfo(getDB().retrieveMapping(user.getCertificateDN()), groupName);
```

Comment:
Yeah - I think retrieveAccount(GridUser) in the pool mapper is better than the retrieveAccount(String) in the manual mapper.